wsd/ClientRequestDispatcher.cpp

```diff
@@ -1096,6 +1101,247 @@ bool ClientRequestDispatcher::handleWopiDiscoveryRequest(
     return true;
 }
 
+STATE_ENUM(CheckStatus,
```

Comment:
Lets have a comment here stating: "/// NB. These names are part of the published API - and should not be renamed or altered but can be expanded"

Reply:
with clang  it stops the build due to unused functions:
```
wsd/ClientRequestDispatcher.cpp:1106:1: error: unused function 'toStringShort' [-Werror,-Wunused-function]
STATE_ENUM(CheckStatus,
^
./common/StateEnum.hpp:63:31: note: expanded from macro 'STATE_ENUM'
    static inline std::string toStringShort(NAME e)                                                \
                              ^
wsd/ClientRequestDispatcher.cpp:1106:1: error: unused function 'toString' [-Werror,-Wunused-function]
./common/StateEnum.hpp:76:31: note: expanded from macro 'STATE_ENUM'
    static inline std::string toString(NAME e)                                                     \
                              ^
wsd/ClientRequestDispatcher.cpp:1106:1: error: unused variable 'CheckStatusMax' [-Werror,-Wunused-const-variable]
./common/StateEnum.hpp:80:25: note: expanded from macro 'STATE_ENUM'
    static const size_t NAME##Max = COUNT_ARGS(__VA_ARGS__);                                       \
                        ^
<scratch space>:107:1: note: expanded from here
CheckStatusMax
^
3 errors generated.

```

Reply:
fix: https://github.com/CollaboraOnline/online/pull/10785